### ISSUE TITLE
Distribution of maps by mappers and modders

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#1839]
For now, when someone creates a map, port a map from another game (including tremulous), repack a map from tremulous (port != repack, repack is just about moving files around and generating navmeshes), modifies a map, etc, there is no way to easily share the result and keep it up to date.

What I would suggest is a mechanism like a syncthing network.
Why syncthing? Because it's based on both parts exchanging their keys. Unlike git, it's bi-directionnal. Unlike git, it does not shares or stores history (useful for people with small storage space!).
There's likely many other solutions, but to find one, the question must be asked.

I understand it's far from perfect, though, and there's a lot of policy behind. But I think this tool, which wants both public keys of emitter and receiver to do a data exchange, could be a good starting base.

**Comment by necessarily-equal** [comment]
I have some doubts about this.

Can someone that didn't send they key see files at least? And huh, what happens if someone just deletes a file? Or corrupts one by accident? This sounds like it also makes it easier to loose work everywhere.

**Comment by Gireen** [comment]
Whats the benefit compared to a central repository and packet manager?

**Comment by bmorel** [comment]
> Can someone that didn't send they key see files at least?

Of course. We're talking about public keys here. Only needed to send, not to read.

> And huh, what happens if someone just deletes a file? Or corrupts one by accident? This sounds like it also makes it easier to loose work everywhere.

Syncthing does claim to have history feature. But syncthing is just an example for implementation.


> Whats the benefit compared to a central repository and packet manager?

Lower official maintenance. Central stuff requires officially appointed people to maintain it, while my proposal allows informal spreading, outside of official responsibility.
I guess it's kind of near the mesh network proposal...

**Comment by Gireen** [comment]
If it is decentralized and without any direct official control can't this already be used?
Server owners can share their folders with it.

How would the practical use be?
For example with a central repository it would be:

- creators load mods/maps/layouts up to a website
- servers admins add repository to the download urls, pick what they want and download it
- the updater client could have a section for players to manually download any community content


What happens with harmful and illegal content if there is no moderation?
I doubt we would want to display or distribute such things in an official interface.
